File: src/com/train/algorithm/twoPointers/SquaresofaSortedArray.cpp
//
// Created by 罗斯 on 1/9/2022.
//
#include <vector>

using namespace std;
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> res(nums.size());
        int l = 0,r = nums.size() - 1;
        for(int i = nums.size() - 1; i >= 0;--i){
            if(abs(nums[r]) > abs(nums[l])) res[i] = nums[r] * nums[r--];
            else res[i] = nums[l] * nums[l++];
        }
        return res;
    }
};

File: src/com/train/algorithm/twoPointers/implementInPython/SquaresOfASortedArray.py
class Solution(object):
    def sortedSquares(self, nums):
        l,r = 0, len(nums)-1
        res = []
        while l <= r:
            if abs(nums[l]) > abs(nums[r]):
               res.append(nums[l]**2)
               l += 1
            else:
                res.append(nums[r]**2)
                r -= 1

        return res[::-1]